### PR TITLE
Development builders [builders]

### PR DESCRIPTION
**Pull request by andrewayoub**
Resolves: https://github.com/threefoldtech/jumpscaleX/issues/155

## Description

Refactor the most important builders to use DSL for specifying how to sandbox/build/...

## Checklists
 - [x] lua
 - [x] openresty
 - [x] Nginx
 - [x] Caddy
 - [x] ETCD
 - [x] Traefik
 - [x] Coredns
 - [x] HumHub

### Before asking for approval

- [ ] Make sure the changes are covered by tests.
- [ ] All the CI tasks are green.
- [ ] Ensure the branch is up to date with its base branch.
- [x] Corresponding changes to the documentation were made.

### CODE REVIEW

**Comment by ahmadnazih** [comment]
Traefik and Caddy builders are finished with their sandbox and flist creation

**Comment by ranatrk** [comment]
CoreDns, OpenResty, and Lua sandbox functions complete. The builders allow building, sandbox, and flist creation

**Comment by andrewayoub** [comment]
A document about how to create a new builder and an example added [here](https://github.com/threefoldtech/jumpscaleX/blob/development_builders/docs/howto/Create%20a%20new%20Builder.md)

File: Jumpscale/builder/network/BuilderCoreDns.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def sandbox(self,  dest_path="/tmp/builders/coredns", sandbox_dir="sandbox", res
         if self._done_check('sandbox',reset):
             return
         if not self._done_check('build'):
```
Comment (andrewayoub) [inline Jumpscale/builder/network/BuilderCoreDns.py]: don't check for build here

**Pull request merged**